framework/src/Volo.Abp.ObjectExtending/Volo/Abp/Data/HasExtraPropertiesExtensions.cs

```diff
@@ -106,7 +105,20 @@ public static void SetDefaultsForExtraProperties(object source, Type objectType)
                 throw new ArgumentException($"Given {nameof(source)} object does not implement the {nameof(IHasExtraProperties)} interface!", nameof(source));
             }
 
-            ((IHasExtraProperties) source).SetDefaultsForExtraProperties(objectType);
+            ((IHasExtraProperties)source).SetDefaultsForExtraProperties(objectType);
+        }
+
+        public static void SetExtraPropertiesToRegularProperties(this IHasExtraProperties source)
+        {
+            var properties = source.GetType().GetProperties()
+                .Where(x => source.ExtraProperties.ContainsKey(x.Name)
+                            && x.GetSetMethod(true) != null)
+                .ToList();
+
+            foreach (var property in properties)
+            {
+                property.SetValue(source, source.ExtraProperties[property.Name]);
```

Comment:
I think we should remove this property from source.ExtraProperties. No need to duplication. If we define a regular property, we want to use it instead of extra property.
Duplication is dangerous here. Because if we change one of them, the other one won't change.